packages/inter-protocol/src/reserve/assetReserveKit.js

```diff
@@ -0,0 +1,278 @@
+import { Fail } from '@agoric/assert';
+import { AmountMath } from '@agoric/ertp';
+import { makeTracer } from '@agoric/internal';
+import { M, makeScalarBigMapStore, prepareExoClassKit } from '@agoric/vat-data';
+import { atomicTransfer } from '@agoric/zoe/src/contractSupport/atomicTransfer.js';
+import { makeRecorderTopic } from '@agoric/zoe/src/contractSupport/topics.js';
+import { E } from '@endo/eventual-send';
+
+const { quote: q } = assert;
+
+const trace = makeTracer('ReserveKit', false);
+
+/**
+ * @typedef {object} MetricsNotification
+ *
+ * @property {AmountKeywordRecord} allocations
+ * @property {Amount<'nat'>} shortfallBalance shortfall from liquidation that
+ *   has not yet been compensated.
+ * @property {Amount<'nat'>} totalFeeMinted total Fee tokens minted to date
+ * @property {Amount<'nat'>} totalFeeBurned total Fee tokens burned to date
+ */
+
+/**
+ * @param {import('@agoric/vat-data').Baggage} baggage
+ * @param {{
+ * feeMint: ZCFMint<'nat'>,
+ * makeRecorderKit: import('@agoric/zoe/src/contractSupport/recorder.js').MakeRecorderKit,
+ * storageNode: StorageNode,
+ * zcf: ZCF}} powers
+ */
+export const prepareAssetReserveKit = async (
+  baggage,
+  { feeMint, makeRecorderKit, storageNode, zcf },
+) => {
+  trace('prepareAssetReserveKit', [...baggage.keys()]);
+  const feeKit = feeMint.getIssuerRecord();
+  const emptyAmount = AmountMath.makeEmpty(feeKit.brand);
+
+  const makeAssetReserveKitInternal = prepareExoClassKit(
+    baggage,
+    'AssetReserveKit',
+    undefined,
```

Comment:
This contract has a publicFacet, so it's likely to be exposed to customers, and should have guards.